### PR TITLE
feat: setup MongoDB with mevn generate

### PR DESCRIPTION
**Pull request by danivijay**
**What kind of change does this PR introduce?**

A feature

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
No

**Summary**

Contributes to #176 . ie,  adding MongoDB configuration in server.

**Does this PR introduce a breaking change?**

No

**Other information**
This is not a working draft you'll get a basic idea on my thought process. I'm very new to this type of application, and there are some grey areas. Please contact in person if possible so that I can clear my doubts @jamesgeorge007

### CODE REVIEW

File: src/commands/basic/generate.js [hunk 1]
```diff
@@ -107,6 +107,29 @@ const generateFile = async () => {
...
+  );
+
+  // Include a new line to compensate the previous addition
+  serverFile.splice(
+    postImportIndex + 1,
+    0,
+    'import mongoDbInit from "./helpers/db/mongodb.js";',
+    '',
+    'mongodbInit();',
+  );
```
Comment (jamesgeorge007) [inline src/commands/basic/generate.js:132]:
```diff
-  const dbFilePath = `server/src/db/mongodb.js`;
-  fs.copyFile('../../helpers/db', dbFilePath, (err) => {
-    if (err) throw err;
-  });
+  copyDirSync(path.join(templatePath, 'helpers'), 'server');

   const serverFile = fs
-    .readFileSync('./server/src/router.js', 'utf8')
+    .readFileSync('./server/server.js', 'utf8')
     .toString()
     .split('\n');

-  const postImportIndex = serverFile.indexOf(
-    serverFile.find((item) => item === ''),
-  );
+  const postImportIndex = serverFile.findIndex((item) => item === '');
+  // second occurence
+  const requiredIndex = serverFile.indexOf('', postImportIndex + 1);

   // Include a new line to compensate the previous addition
   serverFile.splice(
-    postImportIndex + 1,
+    requiredIndex + 1,
     0,
-    'import mongoDbInit from "./helpers/db/mongodb.js";',
+    'require("./helpers/db/mongodb.js")();',
     '',
-    'mongodbInit();',
   );
+
+  fs.writeFileSync('./server/server.js', serverFile.join('\n'));
```

File: src/helpers/db/mongodb.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+const init = () => {
+  mongoose.connect(process.env.DB_URL).catch((err) => {
```
Comment (jamesgeorge007) [inline src/helpers/db/mongodb.js:4]:
```suggestion
  mongoose
    .connect(process.env.DB_URL, {
      useNewUrlParser: true,
      useUnifiedTopology: true,
    })
    .catch((err) => {
```

File: src/helpers/db/mongodb.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    process.exit(1);
+  });
+  mongoose.connection.on('open', () => {
+    console.log('connected to database');
+  });
+};
+
+mongoose.Promise = global.Promise;
+
+export default init;
```
Comment (jamesgeorge007) [inline src/helpers/db/mongodb.js:15]:
```suggestion
module.exports = init;
```

**Comment by jamesgeorge007** [comment]
Also, please remove these:-

https://github.com/madlabsinc/mevn-cli/blob/568f2810ed855f57e30c5317b0bc605a319f7c4f/src/templates/server/basic/server.js#L5

https://github.com/madlabsinc/mevn-cli/blob/568f2810ed855f57e30c5317b0bc605a319f7c4f/src/templates/server/basic/server.js#L27-L36

File: src/commands/basic/generate.js (hunk 1 above)
Comment (danivijay) [reply src/commands/basic/generate.js]: updated

**Comment by danivijay** [comment]
@jamesgeorge007 all the review comments are addressed and PR is updated. Please re-review.

File: src/commands/basic/generate.js (hunk 1 above)
Comment (jamesgeorge007) [reply src/commands/basic/generate.js]: You forgot to write back the updated file 😅

**Comment by danivijay** [comment]
@jamesgeorge007 added missed file rewrite too. Please verify

**Pull request merged**